metadata-ingestion/src/datahub/ingestion/transformer/pattern_cleanup_ownership.py

```diff
@@ -58,11 +62,19 @@ def transform_aspect(
         # clean all the owners based on the parameters received from config
         cleaned_owner_urns: List[str] = []
         for owner_urn in current_owner_urns:
-            user_id: str = owner_urn.split(_USER_URN_PREFIX)[1]
-            for value in self.config.pattern_for_cleanup:
-                user_id = re.sub(value, "", user_id)
-
-            cleaned_owner_urns.append(_USER_URN_PREFIX + user_id)
+            try:
+                user_id: str = owner_urn.split(_USER_URN_PREFIX)[1]
+                for value in self.config.pattern_for_cleanup:
+                    user_id = re.sub(value, "", user_id)
+                cleaned_owner_urns.append(_USER_URN_PREFIX + user_id)
+            except IndexError:
+                try:
+                    group_id: str = owner_urn.split(_GROUP_URN_PREFIX)[1]
```

Comment:
ideally we'd actually do `Urn.from_string` and parse into proper `CorpUserUrn` / `CorpGroupUrn`, and then do things based on that. And we can catch + ignore the InvalidUrnError

I'm not a huge fan of using the nested try...except clauses for control flow

Reply:
it might also be nice to move this into a helper method e.g. `_process_owner_urn`